tasks/event_tasks.py

```diff
@@ -94,9 +97,30 @@ def random_envelope_event_task_factory(task_params=None):
 
     def inner(user):
         event = event_generator()
+
         project_info = get_project_info(user)
-        envelope = Envelope()
+        # push the current public key in params (this can't be hard coded)
+        header_params = {**task_params, "public_key": project_info.key, "event_id": event.get("event_id")}
```

Comment:
You can skip the `public_key` here. SDKs usually should not send it, and Relay takes the public key from the DSN.

Reply:
I believe without it, tracestate becomes invalid and relay disables sampling.

Reply:
You're right with the trace state. This is not the trace context, though, this is the optional public key in the envelope header.